spec/features/indexing_identifiers_spec.rb

```diff
@@ -16,155 +16,142 @@
-          expect(page).to have_css('dd.blacklight-id', text: solr_id)
-        end
+    it 'matches query with bare and prefixed druid' do
+      [prefixed_druid, prefixed_druid.split(':').last].each do |query|
+        fill_in 'q', with: query
+        click_button 'search'
+        expect(page).to have_content('1 entry found')
+        expect(page).to have_css('dd.blacklight-id', text: solr_id)
       end
     end
+  end
 
-    context 'for sourceids' do
-      # SPEC: Source ID: M2549_2022-259_stertzer, where M2549 is the collection number and 2022-259 is the accession number
-      # sul:M0997_S1_B473_021_0001 (S is for series, B is for box, F is for folder ...)
-      let(:source_id) { "sul:M2549_2022-259_stertzer_#{SecureRandom.alphanumeric(12)}" }
-      let(:item) { FactoryBot.create_for_repository(:persisted_item, source_id:) }
+  context 'for sourceids' do
+    # SPEC: Source ID: M2549_2022-259_stertzer, where M2549 is the collection number and 2022-259 is the accession number
+    # sul:M0997_S1_B473_021_0001 (S is for series, B is for box, F is for folder ...)
+    let(:source_id) { "sul:M2549_2022-259_stertzer_#{SecureRandom.alphanumeric(12)}" }
+    let(:item) { FactoryBot.create_for_repository(:persisted_item, source_id:) }
+
+    it 'matches whole string, including prefix before first colon' do
+      fill_in 'q', with: source_id
+      click_button 'search'
+      # expect a single result, but Solr may not finish commit for previous test delete in time
+      # expect(page).to have_content('1 entry found')
+      expect(page).to have_css('dd.blacklight-id', text: solr_id)
+    end
 
-      before do
-        item.identification.sourceId # ensure item is created before searching
-      end
+    it 'matches without prefix before the first colon' do
+      fill_in 'q', with: source_id.split(':').last
+      click_button 'search'
+      # expect a single result, but Solr may not finish commit for previous test delete in time
+      # expect(page).to have_content('1 entry found')
+      expect(page).to have_css('dd.blacklight-id', text: solr_id)
+    end
 
-      it 'matches whole string, including prefix before first colon' do
-        fill_in 'q', with: source_id
+    it 'matches source_id fragments' do
+      fragments = [
+        'M2549',
+        '2022-259', # accession number
+        'M2549_2022-259',
+        'M2549 2022 259',
+        'stertzer',
+        '259_stertzer',
+        '259-stertzer',
+        '259 stertzer'
+      ]
+      fragments.each do |fragment|
+        fill_in 'q', with: fragment
         click_button 'search'
         # expect a single result, but Solr may not finish commit for previous test delete in time
         # expect(page).to have_content('1 entry found')
         expect(page).to have_css('dd.blacklight-id', text: solr_id)
       end
+    end
 
-      it 'matches without prefix before the first colon' do
-        fill_in 'q', with: source_id.split(':').last
```

Comment:
I moved some stuff around and tried to have fewer test setups;  I didn't ditch any functionality in the tsts.